appserver/ejb/ejb-timer-service-app/src/main/java/com/sun/ejb/timers/TimerWelcomeServlet.java

```diff
@@ -88,7 +85,15 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
             Set persistenttimers = timer.findActiveTimersOwnedByThisServer();
             // Non-persistent timers get directly from the service
 
-            EJBTimerService ejbTimerService = EJBTimerService.getEJBTimerService();
+            NonPersistentEJBTimerService ejbTimerService = null;
+            if (EJBTimerService.isPersistentTimerServiceLoaded()) {
+                ejbTimerService = (NonPersistentEJBTimerService) EJBTimerService.getPersistentTimerService();
+            } else if (EJBTimerService.isPersistentTimerServiceLoaded()) {
+                ejbTimerService = (NonPersistentEJBTimerService) EJBTimerService.getNonPersistentTimerService();
+            }
```

Comment:
This if/else statement irritates me. Is this really meant this way?

Reply:
That's a typo. Thanks for pointing out.